### DISCUSSION TITLE
hosting the mockoon created API

### DISCUSSION THREAD

**Discussion by roybarber** [mockoon/mockoon#576] [Help] [answered]
Hey everyone. 

Has anyone found a solution to running the mockoon CLI in the cloud anywhere. We have a project that deploys to netlify for the static front end build, locally we develop against mockoon endpoints. But stakeholders want to see the working version on netlify, meaning we would need to somehow make the endpoints in mockoon publicly accessible.

Any ideas?

**Comment by 255kb** [comment, marked as answer]
Hi Roy, 
The CLI is designed to be run anywhere in the cloud. However, it's an NPM package requiring a Node.js capable environment, and I think Netlify offers frontend hosting and cloud functions only (but for this use case cloud functions won't be enough). 
From here you have mainly two possibilities: 
- Run it on a VPS (AWS, Digital Ocean, etc), with the port on which Mockoon CLI is running opened. Then you can use the public IP of the VPS + port in your frontend app deployed on Netlify. There is also a [Docker image](https://hub.docker.com/r/mockoon/cli) that you can use to run the CLI on services deploying this kind of containers. 
- Host it on a local computer and use a service like Ngrok to tunnel to it. Then you will have an Ngrok URL redirecting to your local running CLI. You can also implement your own tunnel: https://zach.codes/roll-your-own-ngrok/.

I hope this helps.
